addons/website/static/src/snippets/s_table_of_content/options.js

```diff
@@ -78,9 +78,17 @@ options.registry.TableOfContent = options.Class.extend({
      * @private
      */
     _generateNav: function (ev) {
+        const targetedEls = Array.from(this.$target[0].querySelectorAll(this.targetedElements));
+        const blockTextContent = this.$target[0].textContent.replaceAll('\n', '').trim();
+        if (targetedEls.length === 0 && blockTextContent === '') {
+            // destroy public widget and remove the ToC since there are no more
+            // child elements.
+            this.trigger_up('remove_snippet', {$snippet: this.$target});
+            this.observer.disconnect();
```

Comment:
You disconnect the observer here because otherwise, `_generateNav` can be called before the option is destroyed right ?

Reply:
Not exactly, but when we remove the h1, and h2 and then remove the last block it calls `remove_snippet` even after there's no content left which leads to tb.

Reply:
@Guillaume-gdi
Okay, what is going on here if we don't do `this.observer.disconnect()` is:
- remove h1, h2
- remove block
  - triggers `content_changed` -> this._generateNav()
  - which triggers `'remove_snippet'` -> MutationObserver -> `this._generateNav()`
  - again `remove_snippet` and `tb`

Reply:
By disconnecting the observer, we prevent unnecessary `_genrateNav()` calls which are going to check `targetedEls.length === 0 && blockTextContent === ''` and trigger `remove_snippet` even after we already have removed the snippet.

Reply:
Ok that is what I imagined before
 > You disconnect the observer here because otherwise, _generateNav can be called before the option is destroyed right ?
 
 You can just:
- add a comment in the code to explain why we disconnect the observer
- disconnect the observer before `this.trigger_up('remove_snippet'` to be sure the bug you describe cannot happen

Thanks :+1: